src/main/resources/flyway/sql/V2__new_table_primarysite_cancertype.sql

```diff
@@ -0,0 +1,34 @@
+CREATE TYPE membershiptype AS ENUM ('FULL', 'ASSOCIATE');
+
+CREATE TABLE CANCER (
+    id                UUID PRIMARY KEY,
+    name              VARCHAR(255) UNIQUE NOT NULL
+);
+
+CREATE TABLE SITE  (
+    id                UUID PRIMARY KEY,
+    name              VARCHAR(255) UNIQUE NOT NULL
+);
+
+CREATE TABLE PROGRAMCANCER (
+  program_id          UUID NOT NULL,
+  cancer_id           UUID NOT NULL,
+  PRIMARY KEY(program_id, cancer_id),
+  FOREIGN KEY(program_id) REFERENCES PROGRAM(id),
+  FOREIGN KEY(cancer_id)  REFERENCES CANCER(id)
+);
+
+CREATE TABLE PROGRAMSITE (
+  program_id            UUID  NOT NULL,
+  site_id               UUID  NOT NULL,
+  PRIMARY KEY(program_id, site_id),
+  FOREIGN KEY(program_id) REFERENCES PROGRAM(id),
+  FOREIGN KEY(site_id)  REFERENCES SITE(id)
+);
+
+ALTER TABLE PROGRAM ADD COLUMN date_updated TIMESTAMP NOT NULL;
+ALTER TABLE PROGRAM ADD COLUMN institutions VARCHAR(255) DEFAULT '';
+ALTER TABLE PROGRAM ADD COLUMN countries VARCHAR(255) NOT NULL;
+ALTER TABLE PROGRAM ADD COLUMN regions VARCHAR(255) DEFAULT '';
```

Comment:
according to the spec: ![spec](https://wiki.oicr.on.ca/rest/gliffy/1.0/embeddedDiagrams/c4c803c4-91c7-4704-be27-144cac849582.png)

Regions has a zero-or-many to many programs relationship, meaning its optional
countries has a many to many relationship to programs, and there must be at least one country per program

this means the region and country tables must also be defined, along with their join tables

also, after a discussion with @rosibaj  we have decided to make `submittedDonors` and `genomicDonors` required and mutable, meaning they must be non null, and initialised to 0 upon creation. `Institutions` is a mutable and required field  (i.e non-null) that is a string. Refer to the [spec](https://wiki.oicr.on.ca/display/icgcargotech/Program+Service+-+Backend#ProgramService-Backend-3.1.1.1Program) for more details

after a discussion with @andricDu , we have agreed on a sql naming convention:
- table names should be lowercase, and snake case
- field names should be all lowercase and snake case
- all sql keywords (like `CREATE`, `NOT NULL`) should be uppercase. This will make reading the schema alot easier.

So in summary please do the following changes:
1. add the right constraints for submitted_donors, genomic_donors and institutions
2. create the region and country table and link them to the program
3. update all tables and field names to follow the sql naming convention

Reply:
As discussed with @rosibaj , we decided to keep Countries and Region as text fields in Program table. Will update ERD and specs later.